spec/lib/puppet_spec/compiler.rb

```diff
@@ -28,6 +28,14 @@ def compile_to_relationship_graph(manifest, prioritizer = Puppet::Graph::Sequent
     graph
   end
 
+  def apply_manifest(manifest_content)
+    manifest_path = file_containing("manifest.pp", manifest_content)
+    apply = Puppet::Application[:apply]
+    apply.command_line.args = [manifest_path]
+
+    apply.run
```

Comment:
I think the `run` method will try to exit, so you might need to add something like

```ruby
expect {
  apply.run
}.to exit_with(0)
```